src/classes/ADDR_Addresses_TEST.cls

```diff
@@ -998,6 +998,254 @@ public with sharing class ADDR_Addresses_TEST {
+        addressWithoutCountryState.MailingCountry__c = null;
+        update addressWithoutCountryState;
+
+        System.assertEquals(null, addressWithoutCountryState.MailingState__c);
+        System.assertEquals(null, addressWithoutCountryState.MailingCountry__c);
+
+        Test.startTest();
+        ADDR_Addresses_TDTM.updateAddressCountryAndStateWithAccountOrContactCountryCodeAndStateCode(con, 'Mailing', addressWithoutCountryState);
+        Test.stopTest();
+
+        System.assertEquals('CA', addressWithoutCountryState.MailingState__c);
+        System.assertEquals('United States', addressWithoutCountryState.MailingCountry__c);
+    }
+
+
+    /*********************************************************************************************************
+    @description
+        Update Address' Mailing Country (with State & Country Picklists enabled) when the Country is empty.
+    verify:
+        New Address' Country field is updated with the source sObject's CountryCode value
+    **********************************************************************************************************/
+    static testMethod void testUpdateEmptyAddressMailingCountryWithAccountOrContactCountryCode() {
+        if (!ADDR_Addresses_TDTM.isStateCountryPicklistsEnabled) {
+            return;
+        }
+
+        if (strTestOnly != '*' && strTestOnly != 'testUpdateEmptyAddressMailingCountryWithAccountOrContactCountryCode') {
+            return;
+        }
+
+        Integer hhSize = 1;
+        Integer contactSize = 1;
+
+        // This creates a default Address for each HH
+        createHHTestData(hhSize, contactSize);
+
+        Contact con = listConT[0];
+        con.put('MailingCountryCode', 'US');
+        con.is_Address_Override__c = true;
+
+        Address__c addressWithoutCountry = listAddrT[0];
+        System.assertEquals('United States', listAddrT[0].MailingCountry__c);
+
+        // Set the Mailing Country to null
+        addressWithoutCountry.MailingCountry__c = null;
+        update addressWithoutCountry;
+        System.assertEquals(null, addressWithoutCountry.MailingCountry__c);
+
+        Test.startTest();
+        ADDR_Addresses_TDTM.updateEmptyAddressMailingCountryWithAccountOrContactCountryCode(con, 'Mailing', addressWithoutCountry);
+        Test.stopTest();
+
+        System.assertEquals('United States', addressWithoutCountry.MailingCountry__c);
+    }
+
+
+    /*********************************************************************************************************
+    @description
+        Update Address' Mailing State (with State & Country Picklists enabled) when the State is empty
+        and/or mismatched.
+    verify:
+        New Address' State field is updated with the source sObject's StateCode value
+    **********************************************************************************************************/
+    static testMethod void testUpdateEmptyOrMismatchedAddressMailingStateWithAccountOrContactStateCode() {
```

Comment:
Maybe we can rename the test to match the expected result: `testEmptyAddressStateIsPopulatedFromStateCode`?